eZ/Bundle/EzPublishCoreBundle/FieldType/RichText/Converter/Html5Input.php

```diff
@@ -22,4 +23,27 @@ public function __construct($stylesheet, ConfigResolverInterface $configResolver
         $customStylesheets = $customStylesheets ?: array();
         parent::__construct($stylesheet, $customStylesheets);
     }
+
+    /**
+     * Method checks if document contains any links with '[' or ']' sign an replaces it to its html entities version.
+     *
+     * @param DOMDocument $document
+     *
+     * @return DOMDocument
+     */
+    public function convert(DOMDocument $document)
+    {
+        // EZP-27562 - links containing [ or ] sign results in xml validation error
+        foreach ($document->getElementsByTagName('a') as $link) {
+            $href = $link->getAttribute('href');
+            if ($href !== '') {
+                $href = str_replace('[', '%5B', $href);
+                $href = str_replace(']', '%5D', $href);
```

Comment:
yup, this is good example of usage of `[` and `]` in url.
However php will understand both `http://www.example.com?my_array[]=1&my_array[]=2` (undecoded) and `http://www.example.com?my_array%5B%5D=1&my_array%5B%5D=2` (encoded) the same way (at least on my local apache & php both works well). 

What is more important that example you provided `http://www.example.com?my_array[]=1&my_array[]=2` will fail in current solution so bug we have seems to be more severe that we thought at the beginning